pitop/processing/algorithms/ball_detect.py

```diff
@@ -36,40 +36,60 @@ def __init__(self, color):
         }
         self.match_limit = match_limits[color]
 
-        self.center_points = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
-        self.radius = 0
-        self.angle_from_center = None
+        self._center_points_cv = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
+        self._center = None
+        self._radius = 0
+        self._angle = None
 
-    def clear(self):
-        self.center_points.clear()
+    def no_detection(self):
```

Comment:
`ball.no_detection()` means nothing to me - it signals nothing of what it's going to do.

Reply:
I'd be happier leaving this logic in the actual detect function tbh - I only changed it here because you added the clear() function. I had to change the clear function as it wasn't providing the intended functionality and hence I renamed it